core/src/main/java/nl/nn/adapterframework/ldap/LdapClient.java

```diff
@@ -327,8 +328,12 @@ private String arrayToString(String[] values, String separator) {
 //			context.close();
 //		}
 //    }
-  
+    
     public Set<String> searchRecursivelyViaAttributes(String uid, String baseDn, String attribute) throws NamingException {
+    	return searchRecursivelyViaAttributes(uid, baseDn, attribute, null);
+    }
+    
+    public Set<String> searchRecursivelyViaAttributes(String uid, String baseDn, String attribute, String stopCondition) throws NamingException {
```

Comment:
Wat je hier een 'stopCondition' noemt, is eigenlijk het 'searchTarget'. Ik denk dat je het beter ook zo kan noemen.
Ook graag een regel commentaar/javadoc die de werking ervan beschrijft.

Verder denk ik dat het beter is deze method private te maken, en een andere public method toe te voegen:
``public boolean searchRecursivelyViaAttributesForTargetValue()``
Die kan dan 'true' teruggeven als hij gevonden is, in plaats van een halve lijst van attribute values

Reply:
Hernoemd naar searchTarget, Javadoc toegevoegd.

Methode ook omgezet naar private en een public `searchViaAttributesForTargetValue` toegevoegd die een boolean teruggeeft als searchTarget gevonden is. Deze accepteert als input ook een boolean isRecursiveSearch, waarmee de logica verplaatst is naar LdapClient.